_posts/2024-01-02-opensearch-ai-retrospective.md

```diff
@@ -0,0 +1,205 @@
+
+In order to provide the flexibility to support the many AI use cases and variations that users are asking for, we’re revamping our framework to enable composable pipelines built on configurable and reusable components. For instance, when you integrate models with OpenSearch, we want you to be able to configure the interfaces, so that you’re not limited to integrating text embedding models. We want to enable users to compose a custom query flow as a series of configurable processors, such as custom models—as a generic ML model processor—that enrich query inputs and results during steps in a flow.
+
+We plan to provide templates for precomposed flows to quick-start builders on popular use cases, but leave them with the flexibility to alter and evolve these flows. These templates will capture the OpenSearch resources and configurations required to provision components like search pipelines and AI connectors to power specific use cases like multimodal search. We aim to simplify the backend plumbing required by AI-enhanced search solutions to uplift ease-of-use with minimal flexibility trade-offs.
+
+
+![flow-designer](/assets/media/blog-images/2024-01-02-opensearch-ai-retrospective/flow-designer.png)*Build AI apps faster by composing AI flows through OpenSearch’s no-code experience*([RFC](https://github.com/opensearch-project/OpenSearch-Dashboards/issues/4755))
+
+
+### **New AI Use Case Support**
+
+With the composable flow paradigm, we envision supporting the existing use cases, variations and a plethora of new possibilities. Among the possible use cases will be the ability to construct re-ranking workflows. You will be able to compose a workflow that can retrieve results through lexical or vector methods and efficiently send those results to a re-ranking model, which might be a custom model or a provider like Cohere ReRank. These workflows will unlock various use cases, including search relevancy boosting through state-of-the-art techniques like combining bi-encoder and cross-encoder models and user-level personalization by re-ranking results using deep recommender models. 
+
+As well, the ability to use the generic ML processor within last-mile ingestion pipelines, will unlock countless use cases based around automating metadata extraction to enrich indexes. We plan to provide you with an ingest processor that allows you to configure your model interface, so that you can more easily integrate models ranging from name-entity-recognition models, computer vision classifiers, sentiment detectors to neural machine translators to enrich your data during ingestion and indexing flows.
```

Comment:
```suggestion
Additionally, the ability to use the generic ML processor within last-mile ingest pipelines will unlock countless use cases related to automating metadata extraction in order to enrich indexes. We plan to provide you with an ingest processor that allows you to configure your model interface so that you can more easily integrate models, including named-entity recognition models, computer vision classifiers, sentiment detectors, and neural machine translators, to enrich your data during ingestion and indexing flows.
```